packages/app/src/Charcoal/App/ServiceProvider/EventServiceProvider.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Charcoal\App\ServiceProvider;
+
+use Charcoal\App\Event\EventDispatcher;
+use Charcoal\App\Event\EventListenerInterface;
+use Charcoal\Factory\FactoryInterface;
+use Charcoal\Factory\GenericFactory;
+use Pimple\Container;
+use Pimple\ServiceProviderInterface;
+use Psr\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Event Service Provider. Configures and provides a PDO service to a container.
+ */
+class EventServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * @param Container $container A service container.
+     * @return void
+     */
+    public function register(Container $container)
+    {
+        /**
+         * @param Container $container Pimple DI container.
+         * @return EventDispatcherInterface
+         */
+        $container['event/dispatcher'] = function (Container $container): EventDispatcherInterface {
+            $dispatcher = new EventDispatcher();
+            $dispatcher->setLogger($container['logger']);
+
+            foreach ($container['event/listeners'] as $event => $listeners) {
+                if (is_string($listeners)) {
+                    $listeners = [$listeners];
+                }
```

Comment:
Shouldn't this be formatted as an associative array given the following loop?

```suggestion
                if (is_string($listeners)) {
                    $listeners = [ $listeners => [] ];
                }
```

Reply:
Indeed thanks